app/policies/story_policy.rb

```diff
@@ -8,4 +15,24 @@ def resolve
       end
     end
   end
+
+  def new?
+    @user.present?
+  end
+
+  def create?
+    @user.present?
+  end
+
+  def edit?
+    @user.admin? || @user.editor? || (@user.author? && @user == @story.user)
+  end
+
+  def update?
+    @user.admin? || @user.editor? || (@user.author? && @user == @story.user)
```

Comment:
Nice. 
`@user.author? && @user == @story.user`
This line can be in separate method for  better readability. For example called: users_story?

But its just a improvement, I ll leave it up to you.